homeassistant/components/goalzero/switch.py

```diff
@@ -52,10 +52,10 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn off the switch."""
         payload = {self.entity_description.key: 0}
         await self._api.post_state(payload=payload)
-        self.coordinator.async_set_updated_data(data=payload)
+        self.coordinator.async_set_updated_data(None)
```

Comment:
The `data` attribute is never accessed directly, always via `api`, and the return type for `_async_update_data` is `None`.